### PR TITLE
(FIT-354) - updated Jenkins Shared Library to allow for projects with multiple artifacts

### PR DESCRIPTION
**Pull request by SergiiRedchuk**
https://osi-cwds.atlassian.net/browse/FIT-354

### CODE REVIEW

File: src/gov/ca/cwds/jenkins/semver/TagFetcher.groovy [hunk 1]
```diff
@@ -7,11 +7,24 @@ class TagFetcher {
...
   }
 
-  def getTags() {
+  def getTags(tagPrefix = '') {
     def rawTags = script.sh(script: "git tag", returnStdout: true)
-    def list = rawTags.split("\n").findAll { it =~ /(\d+\.\d+\.\d+)/ }
+    def regex = tagPrefix ? /$tagPrefix\-(\d+\.\d+\.\d+)/ : /(\d+\.\d+\.\d+)/
+    def tags = extractSemVerTags(rawTags, regex)
+    if (tagPrefix && !tags) {
+      // yet no tags with the prefix but could be tags with just versions (no other prefixes) to use further
```
Comment (edgibbs) [inline src/gov/ca/cwds/jenkins/semver/TagFetcher.groovy]: This a question about how we really want to handle this, if so the comment could be removed to make a private method with a meaningful name.

File: test/gov/ca/cwds/jenkins/semver/SemVerSpecification.groovy [hunk 2]
```diff
@@ -19,7 +21,7 @@ class SemVerSpecification extends Specification {
     semVer.newTag('minor')
 
     then:
-    1 * tagFetcher.getTags() >> ["1.1.1", "0.3.4"]
+    1 * tagFetcher.getTags(_) >> ["1.1.1", "0.3.4"]
```
[the remark this answers is not in the document]
Comment (SergiiRedchuk) [reply test/gov/ca/cwds/jenkins/semver/SemVerSpecification.groovy]: fixed

File: test/gov/ca/cwds/jenkins/semver/SemVerSpecification.groovy [hunk 3]
```diff
@@ -40,9 +42,48 @@ class SemVerSpecification extends Specification {
     semVer.newTag('')
 
     then:
-    1 * tagFetcher.getTags() >> ["1.1.1", "0.3.4"]
+    1 * tagFetcher.getTags(_) >> ["1.1.1", "0.3.4"]
```
[the remark this answers is not in the document]
Comment (SergiiRedchuk) [reply test/gov/ca/cwds/jenkins/semver/SemVerSpecification.groovy]: fixed

File: test/gov/ca/cwds/jenkins/semver/SemVerSpecification.groovy [hunk 4]
```diff
@@ -40,9 +42,48 @@ class SemVerSpecification extends Specification {
...
+
+  def "#newTag with empty increment label and a tag prefix"() {
+    given: "a PR with an increment label and a tag prefix label"
+    def pullRequestEvent = Mock(PullRequestEvent)
+    pullRequestEvent.getEvent() >> [labels: [[name: 'minor'], [name: 'lis']]]
+
+    and: "a new Semver"
+    def pipeline = Stub(PipeLineScript)
+    // only labels prefixed with 'lis-' will be taken into account because of the label in PR
+    pipeline.sh(_) >> "cwscms-1.0.4\nlis-0.3.4\nlis-0.1.2"
```
[the remark this answers is not in the document]
Comment (SergiiRedchuk) [reply test/gov/ca/cwds/jenkins/semver/SemVerSpecification.groovy]: fixed

File: test/gov/ca/cwds/jenkins/semver/SemVerSpecification.groovy [hunk 5]
```diff
@@ -40,9 +42,48 @@ class SemVerSpecification extends Specification {
...
+    semVer.pullRequestEvent = pullRequestEvent
+
+    when: "expecting an increment label and one of the tag prefix labels in the PR"
+    def newTag = semVer.newTag('', ['cwscms', 'lis', 'capu'])
+
+    then:
+    newTag == 'lis-0.4.0'
+  }
+
+  def "#newTag with passed increment label and a tag prefix"() {
```
[the remark this answers is not in the document]
Comment (SergiiRedchuk) [reply test/gov/ca/cwds/jenkins/semver/SemVerSpecification.groovy]: So, we discussed the context and communication with leads regarding the approach is in progress.

File: test/gov/ca/cwds/jenkins/semver/TagFetcherSpecification.groovy [hunk 6]
```diff
@@ -50,6 +50,32 @@ class TagFetcherSpecification extends Specification {
...
+
+    when:
+    def tags = tagFetcher.getTags('lis')
+
+    then:
+    tags == ["1.6.0", "1.6.1"]
+  }
+
+  def "#getTags with SemVer tags filtered by version prefix when yet there are no such tags"() {
+    given:
```
[the remark this answers is not in the document]
Comment (SergiiRedchuk) [reply test/gov/ca/cwds/jenkins/semver/TagFetcherSpecification.groovy]: So, we discussed the context and communication with leads regarding the approach is in progress.

File: src/gov/ca/cwds/jenkins/semver/TagFetcher.groovy (hunk 1 above)
Comment (SergiiRedchuk) [reply src/gov/ca/cwds/jenkins/semver/TagFetcher.groovy]: For now, can't figure out how to name the function.
Might not be needed if other approach will be confirmed.

**Pull request merged**